controllers/userController.js

```diff
@@ -1,50 +1,77 @@
+import { sign } from "jsonwebtoken";
+import { indexOf } from "lodash";
 
-import * as models from './../models'
-import { sign } from 'jsonwebtoken'
+export const findAll = (parent, args, context, info) => {
 
-export const findAll = () => {
-  return models.User.findAll()
-}
+  if(!context.user){
+    return [{ id : 404, username: "error no logueado" }]
+  }
+  return context.models.User.findAll();
+};
 
-export const getById = (obj, args) => {
-  return models.User.findByPk(args.id)
-}
+export const getById = (obj, args, context, info) => {
+  return context.models.User.findByPk(args.id);
+};
 
-export const createUser = (obj, { firstname, lastname, username, salt, password }) => {
-  return models.User.create({
+export const createUser = (
+  obj,
+  { firstname, lastname, username, salt, password },
+  context
+) => {
+  return context.models.User.create({
     firstname,
     lastname,
     username,
     salt,
-    password
-  })
-}
+    password,
+  });
+};
 
-export const signUp = async (obj, { data }) => {
-  const user = await SearchByUsername(data.username)
+export const signUp = async (obj, { data }, context) => {
+  const user = await SearchByUsername(data.username, context);
 
   if (user) {
-    return { user: null, accessToken: null, authError: 'usuario ya creado' }
+    return { user: null, accessToken: null, authError: "Usuario ya creado" };
   } else {
-    const newUser = await models.User.create(data)
-    const { id, username } = newUser
-    const accessToken = sign({ id, username }, 'secret', { expiresIn: '10d' })
-    return { user: newUser, jwt: accessToken, authError: null }
+    const newUser = await context.models.User.create(data);
+    const { id, username } = newUser;
+    const accessToken = await sign({ id, username }, "secret", {
+      expiresIn: "10d",
+    });
+    return {
+      user: newUser,
+      jwt: accessToken,
+      authError: null,
+    };
   }
-}
+};
 
-export const signIn = async (obj, { data }) => {
-  const user = await SearchByUsername(data.username)
+export const signIn = async (obj, { data }, context) => {
+  const user = await SearchByUsername(data.username, context);
 
-  if (user) {
-    const { id, username } = user
-    const accessToken = sign({ id, username }, 'secret', { expiresIn: '10d' })
-    return { user: user, jwt: accessToken, authError: null }
+  // if (user && user.password === data.password) {
+  if (user && user.passwordMatches(data.password)) {
+    const { id, username } = user;
+    const accessToken = sign({ id, username }, "secret", { expiresIn: "10d" });
+    return {
+      user: user,
+      jwt: accessToken,
+      authError: null,
+    };
+  }
```

Comment:
I'm not sure if this is a copy & paste, but you should only use `passwordMatches` once to validate the password like this:

```suggestion
  if (!user) return { authError: 'User not found' }
  
  if (!user.passwordMatches(data.password)) return { authError: 'Wrong password' }
  
    const { id, username } = user;
    const jwt = sign({ id, username }, "secret", { expiresIn: "10d" });
    
    return { user, jwt };
  }
```
Then the following part from lines 62 to 66 are not necessary.

Reply:
My bad, I pasted the function as parameter but I didn't  review the hole function for a refactoring! That's something I have to work on!